src/external_cli/arg_builder.rs

```diff
@@ -0,0 +1,115 @@
+#[derive(Debug, Clone)]
+
+/// A helper to make passing arguments to [`std::process::Command`] more convenient.
+pub struct ArgBuilder(Vec<String>);
+
+impl ArgBuilder {
+    /// Create a new builder for command arguments.
+    pub fn new() -> Self {
+        Self(Vec::new())
+    }
+
+    /// Add a singular argument.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_cli::external_cli::arg_builder::ArgBuilder;
+    /// ArgBuilder::new().arg("--release");
+    /// ```
+    pub fn arg<A>(mut self, arg: A) -> Self
+    where
+        A: Into<String>,
+    {
+        self.0.push(arg.into());
+        self
+    }
+
+    /// Add an argument with a value.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_cli::external_cli::arg_builder::ArgBuilder;
+    /// ArgBuilder::new().add_with_value("--bin", "bevy");
+    /// ```
+    pub fn add_with_value<A, V>(self, arg: A, value: V) -> Self
+    where
+        A: Into<String>,
+        V: Into<String>,
+    {
+        self.arg(arg).arg(value)
+    }
+
+    /// Add a boolean flag with the given name, if `value` is `true`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_cli::external_cli::arg_builder::ArgBuilder;
+    /// let is_release = true;
+    /// ArgBuilder::new().add_flag_if("--release", is_release);
+    /// ```
+    pub fn add_flag_if<N>(self, name: N, value: bool) -> Self
+    where
+        N: Into<String>,
+    {
+        if value {
+            self.arg(name)
+        } else {
+            self
+        }
+    }
+
+    /// Add an argument with an optional value.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_cli::external_cli::arg_builder::ArgBuilder;
+    /// let maybe_name = Some("bevy");
+    /// ArgBuilder::new().add_opt_value("--bin", &maybe_name);
+    /// ```
+    pub fn add_opt_value<N, V>(self, name: N, value: &Option<V>) -> Self
```

Comment:
Suggestion: let add_value take `impl Into<Option<V>>` and remove this method

Reply:
That's cool, I like it!

Reply:
Ah, unfortunately it doesn't seem to work well with type interference... Probably because of the nested `Into` definitions.
I also tried `impl Into<Option<String>>`, but that doesn't seem to work with `&str`